orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -402,6 +403,106 @@ public Task taskById(String taskId) {
     }
   }
 
+  /**
+   * Computes all ancestor stages, including those of parent pipelines
+   *
+   * @return list of ancestor stages
+   */
+  public List<Stage> ancestorsWithParentPipelines() {
+    return Stage.getAncestors(this, this.execution);
+  }
+
+  private static List<Stage> getAncestors(Stage stage, Execution execution) {
```

Comment:
note: this is basically a conversion from groovy (from DeploymentDetailsAware.groovy) and move into the `Stage` so it can be reused by both findimage and bake.